R/convertFlowCore.R

```diff
@@ -0,0 +1,177 @@
+library("stats")
+
+#' Convert a gate to flowCore
+#'
+#' Converts a CellEngine gate ("PolygonGate", "EllipseGate", or "RectangleGate") to
+#' its flowCore analogue.
+#'
+#' @param gate The CellEngine gate to be converted
+#' @export
+convertToFlowCore <- function (gate) {
+  if (!requireNamespace("flowCore")) {
+    stop("These utilities require the 'flowCore' package.")
+  }
+
+  switch(
+    gate$type,
+    "RectangleGate" = flowCore::rectangleGate(
+      filterId = gate$name,
+      stats::setNames(list(
+        c(gate$model$rectangle$x1, gate$model$rectangle$x2),
+        c(gate$model$rectangle$y1, gate$model$rectangle$y2)),
+      c(gate$xChannel, gate$yChannel))),
+    "EllipseGate" = {
+      ellipse <- gate$model$ellipse
+      x <- unlist(ellipse$center)[1]
+      y <- unlist(ellipse$center)[2]
+      points <- getEllipsePoints(ellipse$angle, ellipse$major, ellipse$minor, x, y)
+      points <- t(data.frame(points))
+
+      result <- fitEllipsePoints(points)
+
+      cov <- result$covar
+      colnames(cov) <- c(gate$xChannel, gate$yChannel)
+      rownames(cov) <- c(gate$xChannel, gate$yChannel)
+
+      return(flowCore::ellipsoidGate(filterId = gate$name, .gate = cov, mean = c(result$x, result$y)))
+    },
+    "PolygonGate" = {
+      m <- gate$model$polygon$vertices
+      if (is.null(dim(m))) {
+        m <- m[[1]]
+      }
+      colnames(m) <- c(gate$xChannel, gate$yChannel)
+      flowCore::polygonGate(filterId = gate$name, m)
+    },
+    "QuadrantGate" = stop("This gate has no representation in flowCore"),
+    "SplitGate" = stop("This gate has no representation in flowCore")
+  )
+}
+
+#' Scale a flowCore gate or flowDensity filter
```

Comment:
Polygon* gate specifically?

Reply:
It actually scales any gate with two channels just fine, but it doesn't seem to be necessary now